scipy/interpolate/tests/test_fitpack2.py

```diff
@@ -325,6 +325,33 @@ def test_array_like_input(self):
         assert_allclose(spl1([0.1, 0.5, 0.9, 0.99]),
                         spl2([0.1, 0.5, 0.9, 0.99]))
 
+    def test_fpknot_oob_crash(self):
+        # https://github.com/scipy/scipy/issues/3691
+        x = range(109)
+        y = [0.,0.,0.,0.,0.,10.9,0.,11.,0.,
+             0.,0.,10.9,0.,0.,0.,0.,0.,0.,
+             10.9,0.,0.,0.,11.,0.,0.,0.,10.9,
+             0.,0.,0.,10.5,0.,0.,0.,10.7,0.,
+             0.,0.,11.,0.,0.,0.,0.,0.,0.,
+             10.9,0.,0.,10.7,0.,0.,0.,10.6,0.,
+             0.,0.,10.5,0.,0.,10.7,0.,0.,10.5,
+             0.,0.,11.5,0.,0.,0.,10.7,0.,0.,
+             10.7,0.,0.,10.9,0.,0.,10.8,0.,0.,
+             0.,10.7,0.,0.,10.6,0.,0.,0.,10.4,
+             0.,0.,10.6,0.,0.,10.5,0.,0.,0.,
+             10.7,0.,0.,0.,10.4,0.,0.,0.,10.8,0.]
```

Comment:
This needs spaces after the comments to make linting happy

Reply:
are you sure? i've run `flake8` on this on it doesn't complain. there are also lists in the same file that look just like this. what command is used by the linter?

Reply:
https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=18431&view=logs&j=fdd89725-f657-5341-7d77-3c5c44c4f5b6&t=325fc03e-e0b9-5339-adaf-099a49a14600&l=12

Reply:
The linting command is [here](https://github.com/scipy/scipy/blob/main/azure-pipelines.yml#L158)

Reply:
@nkaretnikov, yes, the spaces are needed.  Currently the test suite checks the *new* code with a linter, because there is a lot of legacy code that is not conformant, and there has been resistance to a big reformatting of all that old code.  The checks help ensure that we don't make things worse.

We also have a file `tox.ini` that sets the default parameters for `flake8`, and it is currently set up to be somewhat lenient.  I often run `flake8` with the option `--isolated` on the files that I am editing, so it ignores `tox.ini` and uses all its default rules.  For some of our code, `flake8` will then report many issues; I mainly check that I haven't introduced any new ones.

Reply:
fixed